### PR TITLE
first draft

### PR DESCRIPTION
**Pull request by dominykas**

### CODE REVIEW

**Comment by kestux** [comment]
it's gonna be OS`ed?

**Comment by dominykas** [comment]
It already is a public repo and on npm.

File: index.js [hunk 1]
```diff
@@ -1,5 +1,49 @@
...
-module.exports.decryptHash = function () {
-	throw new Error("Not implemented");
-};
+var forge = require("node-forge");
+
+function trimZeroes(data) {
+	return data.toString().replace(/^\0+/, "").replace(/\0+$/, "");
+}
+
+function padKey(key) {
```
Comment (kestux) [inline index.js:9]: isn't there a lib for that?

File: test/index.test.js [hunk 2]
```diff
@@ -9,8 +9,46 @@ describe("mistertango", function () {
...
+
+		it("should trim encrypted hash", function (done) {
+			mistertango.decryptHash("   myC4pE/gApa74oUJYL9vZDbFnpzw7fUTWPQScobxn+4=   ", "test-key-with-length-23", function (err, msg) {
+				expect(err).to.be(null);
+				expect(msg).to.eql({ok: true});
+				done();
+			});
+		});
+
+		it("should not release zalgo on success", function (done) {
```
Comment (kestux) [inline test/index.test.js:28]: what's zalgo?

Comment (dominykas) [inline test/index.test.js:28]: http://lmgtfy.com/?q=release+zalgo

File: index.js (hunk 1 above)
Comment (dominykas) [inline index.js:9]: there probably is, but I really want to have all of this code in one place and it's simple enough to not care about

File: test/index.test.js (hunk 2 above)
Comment (kestux) [inline test/index.test.js:28]: yeh, I've googled it, but still I don't get it

Comment (dominykas) [inline test/index.test.js:28]: Read the full blog post.

**Comment by kestux** [comment]
done

**Pull request merged**